### ISSUE TITLE
ignoring existing `<picture>` elements, or recognizing an attribute for specific `<img>` elements to ignore

### ISSUE THREAD

**Issue by lewisje** [yireo/Yireo_Webp2#20]
* Brave browser, Version 1.2.43 Chromium: 79.0.3945.130 (Official Build) (64-bit), on Windows 10
* Magento 2.3.4 (latest at this time), PHP 7.1.33
* module version 0.4.6 (latest at this time)
* installed via Composer, first with `composer require yireo/magento2-webp2 --no-update` and then with `composer update`
* I took the usual steps to enable the module: `bin/magento setup:upgrade`, clearing static files, `bin/magento setup:static-content:deploy -f`, clearing cache and compiled code, then `bin/magento setup:di:compile`, all through a couple of Bash scripts that I made so I wouldn't need to keep remembering all of the steps to take.
* This test was actually on a dev environment where there is no cache, but the script I wrote runs the `cache:flush` command anyway; also, because this was in Developer mode, I did not need to use the `module:enable` command to enable the module.
* Under Stores > Configuration > Yireo > Yireo WebP2:
 * Value of Enabled: Yes
 * Value of Debug log: No
 * Value of GD support: WebP is supported by GD
 * Value of PHP version: 7.1.33
 * Value of Module version: 0.4.6
* `/bin/cwebp` version 0.3.0
 * I tried the suggested binary, version 1.0.3, placing it in `~/bin`, but `/bin` takes precedence; I don't have the ability to place executables in `/usr/bin` or a similar places outside the home directory.
* I don't think a demo is necessary, because the module does appear to be working as advertised, and I'm just asking for an improvement; I actually did have links to the dev site and the live site, but then I changed both after I decided to just go with the huge-hero-header flow.
* A screenshot of the source to a *product* page is irrelevant, but I can show you how the section in question is authored vs. how the module alters it (identifying details redacted):

From the main site (okay, the *actual* authoring involved PHP templates, the `store media_url` type):

    <div>
      <a href="https://live.example/products/books.html">
        <picture>
          <source srcset="https://live.example/media/wysiwyg/specials/banner_image_1706.jpg" media="(min-width: 1920px)">
          <source srcset="https://live.example/media/wysiwyg/specials/banner_image_1466.jpg" media="(min-width: 1680px)">
          <source srcset="https://live.example/media/wysiwyg/specials/banner_image_1350.jpg" media="(min-width: 1440px)">
          <source srcset="https://live.example/media/wysiwyg/specials/banner_image_1140.jpg" media="(min-width: 1200px)">
          <source srcset="https://live.example/media/wysiwyg/specials/banner_image_940.jpg" media="(min-width: 992px)">
          <img style="margin:0 auto" src="https://live.example/media/wysiwyg/specials/banner_image_720.jpg" alt="description_of_banner" title="a11y ally">
        </picture>
      </a>
    </div>

From the dev site:

    <div>
      <a href="https://dev.example/products/books.html">
        <picture>
          <source srcset="https://dev.example/media/wysiwyg/specials/banner_image_1706.jpg" media="(min-width: 1920px)">
          <source srcset="https://dev.example/media/wysiwyg/specials/banner_image_1466.jpg" media="(min-width: 1680px)">
          <source srcset="https://dev.example/media/wysiwyg/specials/banner_image_1350.jpg" media="(min-width: 1440px)">
          <source srcset="https://dev.example/media/wysiwyg/specials/banner_image_1140.jpg" media="(min-width: 1200px)">
          <source srcset="https://dev.example/media/wysiwyg/specials/banner_image_940.jpg" media="(min-width: 992px)">
          <picture alt="description_of_banner_without_apostrophes">
            <source type="image/webp" srcset="https://dev.example/media/wysiwyg/specials/banner_image_720.webp">
            <source type="image/jpg" srcset="https://dev.example/media/wysiwyg/specials/banner_image_720.jpg">
            <img style="margin:0 auto" src="https://dev.example/media/wysiwyg/specials/banner_image_720.jpg" alt="description_of_banner" title="a11y ally">
          </picture>
        </picture>
      </a>
    </div>

The issue is that the `<picture>` element that I had written for the main page, crafted to serve a suitably large image depending on browser width to avoid excessive stretching or bandwidth usage, is mangled into something that just shows the lowest-resolution image of the bunch; I also did happen to make suitably named WebP versions of all of those banner images, and I had set up my `.htaccess` file years ago to send the format to browsers that support it, although I could only pre-make WebP versions for pre-rendered one-offs like the front-page banner.

Although I *could* add the `webp_skip` layout handle to the Home Page and any other affected CMS pages, I worry that would keep *any* images on the page from getting the WebP treatment, and I would like to be able to add *some* sort of attribute to the innermost `<img>` element to let the module know to ignore it; an even better idea would be to change the logic of the `ReplaceTags` class to skip over any `<img>` element already inside a `<picture>`, and I might do just that locally via a module that extends that class and uses a `preference`.

In short, the module *does* work for me, but I just wish I could more granularly tell it to ignore certain images.

---
(I have also verified, as the code would imply, that any `srcset` attribute on the `<img>` element, like to specify different images for different device-pixel ratios, is ignored: The generated `<source>` elements are based entirely on the `src` attribute.)

**Comment by jissereitsma** [comment]
Thanks for the thorough explanation. But in short, I don't think it would be good to have further granularity to target specific images. Instead, the problem seems to be that the module turns `img`  tags into `picture` tags, even when the `img` is already contained in a `picture` tag. This is now fixed in the following commit https://github.com/yireo/Yireo_Webp2/commit/26907cfc4a30fd1ee65cd413d88eb8a0db895f79 and this will be shipped in the next release of this module (hopefully towards the end of this week).

**Comment by jissereitsma** [comment]
Version 0.5.1 has now been released.
